$OpenBSD: patch-template_builtin_binary_inc_tmpl,v 1.1.1.1 2022/01/04 19:46:40 jeremy Exp $

Ensure proper alignment of builtin binary arrays to fix crash on mips64.
compile.c accesses the start of these arrays as struct ibf_header.

Index: template/builtin_binary.inc.tmpl
--- template/builtin_binary.inc.tmpl.orig
+++ template/builtin_binary.inc.tmpl
@@ -6,7 +6,7 @@
 %   ary = RubyVM.enum_for(:each_builtin).to_a
 %   ary.each{|feature, iseq|
 
-static const unsigned char <%= feature %>_bin[] = {
+static const unsigned char <%= feature %>_bin[] __attribute__((aligned(8))) = {
 %     iseq                   \
 %     . to_binary            \
 %     . each_byte            \
